worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala

```diff
@@ -333,9 +334,20 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
           logDebug(s"location $location set disk hint to ${location.getStorageInfo} ")
           return fileWriter
         } catch {
-          case t: Throwable =>
+          case fe: FileAlreadyExistsException =>
+            logError(s"Target file ${file.getAbsolutePath} of fileWriter already exists.")
+            exception = new IOException(fe)
+            deviceMonitor.reportNonCriticalError(mountPoint, fe)
+          case rsse: RssException =>
```

Comment:
If we create files failed  just caused by 
```
1. Not enough file handler
2. Not enough space
```
We should not report the critical error to cause all fileWriter in this device fail.